### PR TITLE
fix: add dir option support in link <pkg> --global

### PR DESCRIPTION
**Pull request by lvqq**
Fix: 
1. add dir option support in `pnpm link <pkg> --global`
2. remove WARN **undefined has no binaries**

### CODE REVIEW

File: packages/core/src/install/index.ts [hunk 1]
```diff
@@ -1004,7 +1004,7 @@ const _installInContext: InstallFunction = async (projects, ctx, opts) => {
       const projectToInstall = projects[index]
       if (opts.global && projectToInstall.mutation.includes('install')) {
         projectToInstall.wantedDependencies.forEach(pkg => {
-          if (!linkedPackages?.includes(pkg.alias)) {
+          if (pkg.alias && !linkedPackages?.includes(pkg.alias)) {
             logger.warn({ message: `${pkg.alias} has no binaries`, prefix: opts.lockfileDir })
```
Comment (zkochan) [inline packages/core/src/install/index.ts]: but there is still an issue, no? Maybe something else should be printed instead of the alias.

Comment (lvqq) [reply packages/core/src/install/index.ts]: Yes, but the warning `undefined has no binaries` is more confusing.

Comment (zkochan) [reply packages/core/src/install/index.ts]: I mean the pkg has other fields that may be printed instead of the alias
```suggestion
          if (!linkedPackages?.includes(pkg.alias)) {
            logger.warn({ message: `${pkg.alias ?? pkg.pref} has no binaries`, prefix: opts.lockfileDir })
```

**Review by zkochan — APPROVED** [review]

**Comment by zkochan** [comment]
This whole `--global` and `--dir` code is a bit confusing currently. In other commands `--global` is a replacement for `--dir`. `--global` is like running `--dir=<global-dir>`.

We should probably change how packages are linked globally in the next version of pnpm.

File: packages/core/src/install/index.ts (hunk 1 above)
Comment (lvqq) [reply packages/core/src/install/index.ts]: Yeah, I found it's hard to parse alias of `link:xxx` in `parseWantedDependencies`, so it's better to use another field
